posts/2019-08-12-10-tips-creating-robust-serverless-components.md

```diff
@@ -0,0 +1,88 @@
+
+In this article, we will give you comprehensive practice advice (not docs) on creating robust Serverless Components based on our experience building all these components over the past year, to help as you start creating components for your own unique use cases. So let’s dig in..
+
+### 1. Start with the Outcome
+
+Serverless Components is an outcomes framework. Every single component produces a specific outcome. Those outcomes could be low-level (like an S3 bucket), or high level (like a complete cloud storage solution). Before you even start building your component, you need to decide on what that outcome is and how much abstraction you want to provide for the best UX.
+
+Once you know what outcome you’re after, you’ll need to think about your component’s inputs and outputs, as well as what the defaults are. We recommend that you minimize the inputs at first, then you can iteratively add more config if at all needed, simply because removing inputs later is a breaking change, but adding inputs is not.
+
+If at all possible, aim for a zero config component that would use sane defaults to deploy even a demo of that outcome. One example for a zero-config component is the aws-s3 bucket. If you don’t specify any inputs to that component, it’ll just create an accelerated bucket for you with a randomly generated name.
+
+### 2. See Real-World Examples
+
+While the components core API is simple, the cloud provider you’re working with (ie. AWS) might not be super straightforward. Fortunately there are some repeated patterns that we have seen after creating 20+ serverless components. We recommend that you browse through the available component repos on Github and check which one is close to what you’re building. Just open up the repo and check the code and see how everything fits together.
+All component code is open source, and we are striving to keep it clean, simple and easy to follow. After you look around you’ll be able to understand how our core API works, how we interact with external APIs, and how we are reusing other components. Which brings us to the next point…
+
+### 3. Reuse Existing Components
+
+Serverless Components were designed from the ground up to be sharable and reusable. You can compose components together to deliver higher level component with minimum code. For example, the website component uses the aws-s3 component to deploy and upload to the S3 bucket, and it also uses the domains component to setup the custom domain. Likewise, you can reuse the website component, add to it your own frontend code (ie. an entire React app) to create a configurable/reusable landing page. That’s exactly what the chat-app component does.
```

Comment:
Suggestion:  Add code samples.  This article is a big block of text so far.  Always seek to show, don't tell.  Don't tell them composition is easy, show them via a code sample.

Suggestion:  Add links when you reference other components.